crypto/fipsmodule/sha/sha3_test.cc

```diff
@@ -211,6 +211,26 @@ TEST(SHA3Test, NISTTestVectors_SingleShot) {
   });
 }
 
+TEST(KeccakInternalTest, SqueezeOutputBufferOverflow) {
+    EVP_MD_unstable_sha3_enable(true);
+
+    KECCAK1600_CTX ctx;
+    std::vector<uint8_t> out;
+    std::vector<uint8_t> canary(8);
+    std::fill(canary.begin(), canary.end(), 0xff);
```

Comment:
There is support for doing this "natively" with the address sanitizers. You can "poison" a memory region: https://learn.microsoft.com/en-us/cpp/sanitizers/asan-runtime?view=msvc-170#poisoning.

See https://github.com/aws/aws-lc/blob/main/crypto/mem.c for an example how it's used.

Reply:
very nice. is this poisoning only available when run under ASAN? it looks like [the poisoning functions are stubbed out](https://github.com/aws/aws-lc/blob/main/crypto/mem.c#L82) when not run under ASAN, which would reduce our coverage across platforms here...

Reply:
It's a fair point. Stick to your approach for something that is not cpu agnostic.